src/main/java/tc/oc/pgm/util/UsernameResolver.java

```diff
@@ -61,15 +63,10 @@ public static void resolve(UUID id, @Nullable Consumer<String> callback) {
     }
 
     QUEUE.put(id, callback);
-
-    // If the queue is reaching capacity, try to queue an asynchronous resolve
-    if (QUEUE.size() >= 10) {
-      resolveAll();
-    }
   }
 
   private static void resolveAllSync() {
-    if (!LOCK.tryLock()) return;
+    LOCK.tryAcquire();
```

Comment:
why not just using java's synchronized keyword?

Reply:
Lock acquisition here is async. We actually try to acquire the both before the async task is scheduled and before the actual task starts. This is to prevent a dead lock state if someone calls the sync resolve without the async thread.